aplib.net-demo/Assets/Scripts/Aplib/Actions/UnityPathfinderAction.cs

```diff
@@ -0,0 +1,79 @@
+using Aplib.Core;
+using Aplib.Core.Belief;
+using System;
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace Aplib.Integrations.Unity.Actions
+{
+    public class UnityPathfinderAction<TBeliefSet> : Core.Intent.Actions.Action<TBeliefSet>
+        where TBeliefSet : IBeliefSet
+    {
+        protected static readonly Func<TBeliefSet, bool> _alwaysTrue = _ => true;
+
+        public UnityPathfinderAction(
+            Func<TBeliefSet, Transform> objectQuery,
+            Action<TBeliefSet, Vector3> effect,
+            Vector3 location,
+            float heightOffset = 0f,
+            Func<TBeliefSet, bool> guard = null,
+            Metadata metadata = null)
+            : this(objectQuery,
+                location: ConstantLocation(location),
+                effect,
+                heightOffset,
+                guard,
+                metadata
+            )
+        {
+        }
+
+        public UnityPathfinderAction(
+            Func<TBeliefSet, Transform> objectQuery,
+            Func<TBeliefSet, Vector3> location,
+            Action<TBeliefSet, Vector3> effect,
+            float heightOffset = 0f,
+            Func<TBeliefSet, bool> guard = null,
+            Metadata metadata = null)
+            : base(
+                effect: PathfindingAction(objectQuery, location, effect, heightOffset: heightOffset),
+                guard: guard ?? _alwaysTrue,
+                metadata
+            )
+        {
+        }
+
+        protected static Func<TBeliefSet, Vector3> ConstantLocation(Vector3 location) => _ => location;
+
+        private static Action<TBeliefSet> PathfindingAction(Func<TBeliefSet, Transform> objectQuery,
+            Func<TBeliefSet, Vector3> locationQuery,
+            Action<TBeliefSet, Vector3> effect,
+            float speed = 7f,
+            float heightOffset = 0f) => beliefSet =>
+        {
+            Transform transform = objectQuery(beliefSet);
+            Vector3 target = locationQuery(beliefSet);
+
+            NavMeshPath path = new();
+            NavMesh.CalculatePath(transform.position,
+                target,
+                NavMesh.AllAreas,
+                path
+            );
+
+            if (path.corners.Length <= 1)
+            {
+                return;
+            }
```

Comment:
I thought that we would use this style for single line if statements but I can be wrong.
```suggestion
            if (path.corners.Length <= 1) return;
```

Reply:
Rider does not like it when I do this. It doesn't really matter anyway, so I will just leave it.